cli/commands/rewards/fetchevents.ts

```diff
@@ -91,9 +93,17 @@ export default class FetchEvents extends BaseCommand {
 
     progressBar.update(progressBarTotal)
     progressBar.stop()
-    const attestationsFile = `attestation-completed-events-${fromBlock}-${toBlock}.json`
-    const transferFile = `transfer-cusd-events-${fromBlock}-${toBlock}.json`
-    this.outputToFile(attestationsFile, attestationEvents, 'AttestationCompleted events')
-    this.outputToFile(transferFile, transferEvents, 'Transfer events')
+    this.writeEventsToFiles(attestationEvents, 'Attestation Events', eventsFileLimit)
+    this.writeEventsToFiles(transferEvents, 'Transfer Events', eventsFileLimit)
+  }
+
+  writeEventsToFiles = (events: EventLog[], eventTitle: string, limit: number) => {
+    while (events.length > 0) {
```

Comment:
Doesn't this get stuck on an infinite loop here? I don't see we're changing the events length anywhere. I think maybe in the last line of the loop we need to do something similar to `events = events.slice(limit, events.length - 1)`.

Reply:
`const splicedEvents = events.splice(0, limit)` will take out the first `limit` number of entries, until there are no more

Reply:
Oh, you're right, I misunderstood how splice works, didn't realize it modified the original array as well.